addons/web_editor/models/field_html_history_diff.py

```diff
@@ -0,0 +1,188 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import json
+import re
+
+from datetime import datetime
+from odoo import api, fields, models
+from odoo.tools.misc import _format_time_ago
+from difflib import SequenceMatcher
+
+
+# ------------------------------------------------------------
+# History restoration methods
+# ------------------------------------------------------------
+
+
+def restore_one(current_content, diff_list):
+    diff_list.reverse()
+    current_content_list = current_content.split('<')
+
+    for diff_data in diff_list:
+        diff_data_split = diff_data.split('<')
+        diff_split = diff_data_split[0].split('|', 1)
+        diff_lines = diff_split[1].split(',', 1)
```

Comment:
I use the standard python diff lib indeed. But the format is custom to avoid noise and reduce storage space. So It will need some documentation anyways.